src/settings/patronBlocks/Limits/Limits.js

```diff
@@ -0,0 +1,203 @@
+
+    this.setState({ currentPatronGroupLimits });
+  }
+
+  normializeValue = (value) => {
+    if (typeof (value) !== 'number' || value === 0) {
+      return null;
+    }
+
+    return value;
+  }
+
+  getInitialValues = () => {
+    const {
+      patronGroupId: currentPatronGroupId,
+      patronBlockConditions
+    } = this.props;
+    const { currentPatronGroupLimits } = this.state;
+    const initialLimits = {};
+
+    patronBlockConditions.forEach(({ id }) => {
+      const values = currentPatronGroupLimits.filter(({ patronGroupId, conditionId }) => {
+        return conditionId === id && patronGroupId === currentPatronGroupId;
+      })[0];
+
+      if (values?.conditionId) {
+        initialLimits[values?.conditionId] = values?.value;
+      }
+    });
+
+    return initialLimits;
+  }
+
+  saveLimits = (value) => {
+    const {
+      mutator,
+      patronGroupId: currentPatronGroupId,
+    } = this.props;
+    const { currentPatronGroupLimits } = this.state;
+    const promises = [];
+    let limitPromise;
+
+    forIn(value, (limitValue, blockConditionId) => {
+      const foundLimit = currentPatronGroupLimits.find(({ patronGroupId, conditionId }) => {
+        return blockConditionId === conditionId && currentPatronGroupId === patronGroupId;
+      });
+
+      if (foundLimit?.value === limitValue) return;
+
+      const limitValueNumber = parseFloat(limitValue);
+
+      if (foundLimit) {
+        mutator.patronBlockLimitId.update(foundLimit.id);
+        limitPromise = mutator.patronBlockLimits.PUT({
+          ...foundLimit,
+          value: this.normializeValue(limitValueNumber),
+        });
+      } else {
+        limitPromise = mutator.patronBlockLimits.POST({
+          patronGroupId: currentPatronGroupId,
+          conditionId: blockConditionId,
+          value: this.normializeValue(limitValueNumber),
+        });
+      }
+
+      promises.push(limitPromise);
+    });
+
+    return promises;
+  }
+
+  onSubmit = (value) => {
+    const { patronGroup } = this.props;
+    const limits = this.saveLimits(value);
+    const showSuccessMessage = (
+      <SafeHTMLMessage
+        id="ui-users.settings.limits.callout.message"
+        values={{ patronGroup }}
+      />
+    );
+    const showErrorMessage = <FormattedMessage id="ui-users.settings.limits.callout.error" />;
+
+    if (this.callout) {
+      Promise.all(limits)
+        .then(() => this.callout.sendCallout({
+          type: 'success',
+          message: showSuccessMessage
+        }))
+        .catch(() => {
+          this.callout.sendCallout({
+            type: 'error',
+            message: showErrorMessage
+          });
+
+          return Promise.reject();
+        });
+    }
+  }
+
+  render() {
+    const {
+      patronGroup,
+      patronGroupId,
+      patronBlockConditions,
+    } = this.props;
+
+    if (!this._isMounted) return null;
```

Comment:
I am wondering why it is needed. As I see it can be added to every component in that way. Did you have some error messages/warning that made you do it?

Reply:
I have not any errors. I just wanted to prevent possibility of memory leak, it is recommended for components with state, isn't it. Or do you think I should remove `_isMounted`?

Reply:
I think so, usually such things are added on purpose. As I said, in that way we can introduce such things nearly to every component. But I leave it up to you)